cmd/bcpd/app/init.go

```diff
@@ -69,16 +69,16 @@ func GenInitOptions(args []string) (json.RawMessage, error) {
 }
 
 // GenerateApp is used to create a stub for server/start.go command
-func GenerateApp(home string, logger log.Logger, debug bool) (abci.Application, error) {
+func GenerateApp(options *server.Options) (abci.Application, error) {
 	// db goes in a subdir, but "" -> "" for memdb
 	var dbPath string
-	if home != "" {
-		dbPath = filepath.Join(home, "bcp.db")
+	if options.Home != "" {
+		dbPath = filepath.Join(options.Home, "bcp.db")
 	}
 
 	// TODO: anyone can make a token????
 	stack := Stack(nil)
-	application, err := Application("bcp", stack, TxDecoder, dbPath, debug)
+	application, err := Application("bcp", stack, TxDecoder, dbPath, options)
```

Comment:
And only the debug flag here?
If we only need one option, fair enough to pass it alone.
I like *Options as an interface between the (shared) server code and the (app-specific, maybe in different repo) application generation.